python_modules/dagster/dagster/_core/definitions/asset_dep.py

```diff
@@ -0,0 +1,68 @@
+from typing import NamedTuple, Optional, Union
+
+import dagster._check as check
+from dagster._annotations import PublicAttr, experimental
+from dagster._core.definitions.asset_spec import AssetSpec
+from dagster._core.definitions.assets import AssetsDefinition
+from dagster._core.definitions.partition_mapping import PartitionMapping
+from dagster._core.definitions.source_asset import SourceAsset
+
+from .events import (
+    AssetKey,
+    CoercibleToAssetKey,
+)
+
+
+@experimental
+class AssetDep(
+    NamedTuple(
+        "_AssetDep",
+        [
+            ("asset_key", PublicAttr[AssetKey]),
+            ("partition_mapping", PublicAttr[Optional[PartitionMapping]]),
+        ],
+    )
+):
+    """Specifies a dependency on an upstream asset when using AssetSpec.
```

Comment:
I think we will use this through the system, so I don't think we need to say "when using AssetSpec"